### PR TITLE
Add financial calculator checkboxes to Item Extras Editor

### PR DESCRIPTION
**Pull request by nedredmond**
## Summary:

- Add financial calculator checkboxes to Item Extras
- Add an indentation to dependent checkboxes
- Slight refactor to reduce redundant code
- Added item extras story

Issue: LC-1537

Testing strategy:
Check in new story that nothing has changed except new checkboxes added.
Clicking Financial Calculators checkbox should expand three checked options. Unchecking all options unchecks parent checkbox; unchecking parent unchecks options.

### CODE REVIEW

File: packages/perseus-editor/src/item-extras-editor.tsx [hunk 1]
```diff
@@ -20,159 +26,208 @@ class ItemExtrasEditor extends React.Component<Props> {
...
+            !this.props.financialCalculatorTotalAmount &&
+            !this.props.financialCalculatorTimeToPayOff
+        ) {
+            this.props.onChange({
+                financialCalculator: false,
+            });
+        }
+    }
+
     serialize: () => any = () => {
```
Comment (jeremywiebe) [inline packages/perseus-editor/src/item-extras-editor.tsx:52]: **request**: Can you type this `serialize` method result as the `PerseusAnswerArea` type from `@khanacademy/perseus`, please? That will mean you also need to add these new options to that type, but that's a good thing as that type should be the source of truth for what options are available in the answer area.

File: packages/perseus-editor/src/item-extras-editor.tsx [hunk 2]
```diff
@@ -20,159 +26,208 @@ class ItemExtrasEditor extends React.Component<Props> {
...
+                        }}
+                    />
 
-                    <div>
-                        <label>
-                            Show z table (statistics):{" "}
-                            <input
-                                type="checkbox"
-                                checked={this.props.zTable}
+                    {this.props.financialCalculator && (
```
Comment (jeremywiebe) [inline packages/perseus-editor/src/item-extras-editor.tsx:95]: **question**: Are these three sub-types mutually exclusive?  If so, would it make sense for them to be a set of radio buttons instead?

**Review by jeremywiebe — APPROVED** [review]
This looks good.  One request before landing is to address the typing comment for the `serialize` method.

File: packages/perseus-editor/src/item-extras-editor.tsx (hunk 2 above)
Comment (nedredmond) [reply packages/perseus-editor/src/item-extras-editor.tsx:95]: They are not mutually exclusive!

File: packages/perseus-editor/src/item-extras-editor.tsx (hunk 1 above)
Comment (nedredmond) [reply packages/perseus-editor/src/item-extras-editor.tsx:52]: Can do! I wanted to type this but wasn't sure where it was used or how, didn't investigate. Always happy to improve types.

Comment (nedredmond) [reply packages/perseus-editor/src/item-extras-editor.tsx:52]: @jeremywiebe it doesn't seem advisable to import types from a different package. Each package should export its own types, no? This is reinforcing some very difficult coupling between the packages.

It doesn't look like `PerseusAnswerArea` is used anywhere.

Comment (nedredmond) [reply packages/perseus-editor/src/item-extras-editor.tsx:52]: OH I see now. I'll figure something out here.

File: packages/perseus-editor/src/item-extras-editor.tsx (hunk 2 above)
Comment (jeremywiebe) [reply packages/perseus-editor/src/item-extras-editor.tsx:84]: Ah, then this makes sense.  Thanks.

File: packages/perseus-editor/src/item-extras-editor.tsx (hunk 1 above)
Comment (jeremywiebe) [reply packages/perseus-editor/src/item-extras-editor.tsx]: With regards to exporting/importing types. In this case, the `perseus-editor` package is used to build objects that are consumed by Perseus itself.  So in that case I do think it makes sense for the editors to use types from Perseus to ensure we are building the Perseus data objects correctly. 

In the future, I'd love to see these editors run the data through a validation that's exported from Perseus to guarantee the objects are valid before saving them (ie. not _just_ a type check but a runtime validation).

**Pull request merged**